src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts

```diff
@@ -1,95 +1,61 @@
 import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
 import { AbstractControl, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
-import { MatAutocomplete, MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
-import { Select, Store } from '@ngxs/store';
-import { merge, Observable, Subject } from 'rxjs';
-import { debounceTime, distinctUntilChanged, filter, startWith, takeUntil, tap } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { FormValidators } from 'src/app/shared/constants/validation';
 import { Provider } from 'src/app/shared/models/provider.model';
-import { ValidationConstants } from 'src/app/shared/constants/validation';
-import { NO_LATIN_REGEX } from 'src/app/shared/constants/regex-constants';
-import { Constants } from 'src/app/shared/constants/constants';
-import { Codeficator } from 'src/app/shared/models/codeficator.model';
-import { MetaDataState } from 'src/app/shared/store/meta-data.state';
-import { ClearCodeficatorSearch, GetCodeficatorSearch } from 'src/app/shared/store/meta-data.actions';
-
-const defaultValidators: ValidatorFn[] = [
-  Validators.required,
-  Validators.pattern(NO_LATIN_REGEX),
-  Validators.minLength(ValidationConstants.INPUT_LENGTH_1),
-  Validators.maxLength(ValidationConstants.INPUT_LENGTH_60)
-];
-
-const defaultSearchValidators: ValidatorFn[] = [
-  Validators.required,
-  Validators.pattern(NO_LATIN_REGEX),
-  Validators.minLength(ValidationConstants.INPUT_LENGTH_3),
-  Validators.maxLength(ValidationConstants.INPUT_LENGTH_60)
-];
 
 @Component({
   selector: 'app-create-contacts-form',
   templateUrl: './create-contacts-form.component.html',
-  styleUrls: ['./create-contacts-form.component.scss']
+  styleUrls: ['./create-contacts-form.component.scss'],
 })
```

Comment:
how about OnPush  strategy for components?

Reply:
will be implemented in the scope of separate task where OnPush will be provided everywhere